gradio/cli/commands/components/publish.py

```diff
@@ -217,125 +184,81 @@ def _publish(
-        package_name, version = wheel_file.name.split("-")[:2]
+        package_name, _ = wheel_file.name.split("-")[:2]
         with tempfile.TemporaryDirectory() as tempdir:
             shutil.copytree(
                 str(demo_dir),
                 str(tempdir),
                 dirs_exist_ok=True,
             )
-            if source_dir:
+            if upload_source:
                 shutil.copytree(
                     str(source_dir),
                     str(Path(tempdir) / "src"),
                     dirs_exist_ok=True,
                     ignore=_ignore,
                 )
-            reqs_txt = Path(tempdir) / "requirements.txt"
-            reqs_txt.write_text("\n".join(reqs))
-            readme = Path(tempdir) / "README.md"
-            template = ""
-            if upload_source and source_dir:
-                pyproject_toml = parse((source_dir / "pyproject.toml").read_text())
-                keywords = pyproject_toml["project"].get("keywords", [])  # type: ignore
-                keywords = [
-                    k
-                    for k in keywords
-                    if k not in {"gradio-custom-component", "gradio custom component"}
-                ]
-                if keywords:
-                    template = "," + ",".join(keywords)
-                name = pyproject_toml["project"]["name"]  # type: ignore
-                description = pyproject_toml["project"]["description"]  # type: ignore
 
-            readme_text = README_CONTENTS.format(
-                package_name=package_name,
-                version=version,
-                color_from=color_from,
-                color_to=color_to,
-                template=template,
+            shutil.copyfile(
+                str(source_dir / "README.md"), str(Path(tempdir) / "README.md")
             )
 
-            if name and description:
-                readme_text += f"\n\n# Name: {name}"
-                readme_text += f"\n\nDescription: {description}"
-                readme_text += f"\n\nInstall with: pip install {package_name}"
-
-            readme.write_text(readme_text)
-            dockerfile = Path(tempdir) / "Dockerfile"
-            dockerfile.write_text(make_dockerfile(demo_path.name))
-
             api = HfApi()
-            new_space = api.create_repo(
-                repo_id=f"{package_name}",
-                repo_type="space",
-                exist_ok=True,
-                private=False,
-                space_sdk="docker",
-                token=hf_token,
-            )
+            whoami = api.whoami(token=hf_token)
+            repo_id = f"{whoami['name']}/{package_name}"
+            if not api.repo_exists(repo_id=repo_id, repo_type="space"):
+                api.create_repo(
+                    repo_id=package_name,
+                    repo_type="space",
+                    exist_ok=True,
+                    private=False,
+                    space_sdk="gradio",
+                    token=hf_token,
+                )
```

Comment:
This snippet is valid but a few things to know:
- no need to check `whoami` or `repo_exists` as it can be implicit in the repo creation step
- it's best to pass the token once and for all when initializing `HfApi` rather than passing it to each individual calls
- it's best not to pass `private=False` as this is already the default behavior. In the future, we might provide an enterprise setting to set new repos to private by default. However if `private=False` is already the default in Gradio, we will not be able to overwrite it (in `huggingface_hub` the default is `private=None`).

So this snippet can be updated to:
 
```suggestion
            api = HfApi(token=hf_token)
            repo_url = api.create_repo(
                repo_id=package_name,
                repo_type="space",
                exist_ok=True,
                space_sdk="gradio",
            )
            repo_id = repo_url.repo_id
```

(+ remove all `token=hf_token` args below)

Reply:
Or if there's a reason to do the `whoami` + `repo_exists`, would be good to add a comment explaining why (for future ourselves)

Reply:
Thanks @Wauplin for the suggestion. I misunderstood how `create_repo` works. This suggestion is much better.